staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -378,21 +440,48 @@ func (r *crdHandler) serveResource(w http.ResponseWriter, req *http.Request, req
 		if justCreated {
 			time.Sleep(2 * time.Second)
 		}
-		if terminating {
+		if err := crdInfo.waitForStorageVersionUpdate(req.Context()); err != nil {
```

Comment:
please wrap this in a "gate is enabled" check and leave the old code as the default flow.

Reply:
Added a "feature gate enabled" check in waitForStorageVersionUpdate() itself